src/components/editor/editor-pane/editor-pane.tsx

```diff
@@ -62,15 +63,31 @@ const onChange = (editor: Editor) => {
   }
 }
 
+type ClipboardDataFormats = 'text' | 'url' | 'text/plain' | 'text/uri-list' | 'text/html'
+
 interface PasteEvent {
   clipboardData: {
-    files: FileList
+    files: FileList,
+    getData: (format: ClipboardDataFormats) => string
   },
   preventDefault: () => void
 }
 
 const onPaste = (pasteEditor: Editor, event: PasteEvent) => {
-  if (event && event.clipboardData && event.clipboardData.files && event.clipboardData.files.length > 0) {
+  if (!event || !event.clipboardData) {
+    return
+  }
+
+  // Check for pasted tables
+  const pasteText = event.clipboardData.getData('text')
+  if (pasteText && pasteText.includes('\t') && pasteText.includes('\n')) {
```

Comment:
This is a bit unsafe because these conditions could also apply to python code.
How about an "insert as table" action in the toolbar instead?

Reply:
We could also check if the code is pasted in between three backticks to prevent python code pasteing…

Also could we check that the code has the same number of tabs in each line?

Reply:
sry but if we don't have a rock solid proof detection AND a possibility to say "insert as plain text" or "no, that's not what i meant! format as plain text instead (like in word. i love this feature)", i'm against a auto detection of stuff.